text/078-tbc-title.md

```diff
@@ -0,0 +1,508 @@
+- See also this discussion to [allow modifications using only Python/Django](https://github.com/wagtail/wagtail/discussions/8128)
+
+### Why not more React
+
+#### Where React fits
+
+- React is a different paradigm to HTML with JS, instead it reverses the world as JS first and foremost (JSX is just JS under the hood).
+- React is an incredible solution high complexity, large state, components with a high level of interactivity (e.g. Draftail editor) where the inside implementation is meant to be obscured and hard to control by outside intervention.
+- React components are great for a self-contained component, owning its own state where the DOM structure is contained 100% in React.
+- React provides a consistent approach to one way data flow that helps for testing and development.
+
+#### Why React is not a catch all solution
+
+- React elements need to be initialised for any DOM to be added to the browser, the HTML it produces does not exist until React builds it.
+- Harder to 'replace' an implementation with an extended one, while some React components may be class based components that could be extended there is no central and consistent way to provide this 'API'.
+- Not possible to 'dig in' and modify some data (props) that is used.
+- Not possible to use HTML provided by the server in any safe way, if you wanted to leverage any of the existing Django ecosystem like templating, custom form widgets, template includes for somewhere 'inside' the React component it is just not possible.
+- A specific API must be exposed to customise behaviour, and if you want this data from the server we need to reach for something like Telepath to do it.
+- React takes over the whole tree it controls, so everything down the tree must also be React which means reimplementing everything in React and non-React. While it is possible to add a DOM element and attach something to it (like a jQuery initialisation) it is not idea.
+- It is not practical to output Django server rendered HTML inside a React element, Telepath provides a nice abstraction for data though.
+- A good example of the current situation is the new Wagtail admin sidebar, the code was written in 2021 and will become the default in Wagtail 2.16. This has been built in React which means the support to provide [HTML and any additional JavaScript on the menu items has had to be deprecated](https://github.com/wagtail/wagtail/blob/main/docs/releases/2.16.md#deprecated-sidebar-capabilities). The Wagtail team are forced to decide between writing new UI components in jQuery, ad-hoc JS or React and React makes sense in those options but we lose so much flexibility that pure HTML with 'sprinkles of JS provides. This also leads to a situation where the main editor is no longer part of the HTML provided, it is not in the DOM until JS loads and adds it creating a less than ideal experience for our editors on slower browsers. See Use Case 9 for more details.
```

Comment:
I can take this out, I did not mean it to come across as antagonistic but rather an example of recent new code that may have been done differently if we had other options in Wagtail's frontend.

Reply:
I have removed this and been a bit clearer about my mentions of sidebar, I do not mean to be antagonistic but it would be good to note that we have since added back the ability to provide data attributes/aria attributes to the menu item due to some feedback on issues.